tests/integration/targets/aci_vmm_uplink/tasks/main.yml

```diff
@@ -0,0 +1,158 @@
+    - create_uplink.current.0.vmmUplinkP.attributes.uplinkName == "uplink1"
+
+# CREATE VMWARE UPLINK AGAIN TO TEST IDEMPOTENCE
+- name: Create VMWare Uplink again
+  cisco.aci.aci_vmm_uplink:
+    <<: *aci_info
+    domain: ansible_domain
+    uplink_id: 1
+    uplink_name: uplink1
+    state: present
+  register: create_uplink_again
+
+- name: Verify Uplink Creation idempotence
+  assert:
+    that:
+    - create_uplink_again is not changed
+    - create_uplink_again.current.0.vmmUplinkP.attributes.dn == "uni/vmmp-VMware/dom-ansible_domain/uplinkpcont/uplinkp-1"
+    - create_uplink_again.current.0.vmmUplinkP.attributes.uplinkId == "1"
+    - create_uplink_again.current.0.vmmUplinkP.attributes.uplinkName == "uplink1"
+
+# MODIFY VMWARE UPLINK
+- name: Create VMWare Uplink again
+  cisco.aci.aci_vmm_uplink:
+    <<: *aci_info
+    domain: ansible_domain
+    uplink_id: 1
+    uplink_name: updated_uplink
+    state: present
+  register: update_uplink
+
+- name: Verify Uplink update
+  assert:
+    that:
+    - update_uplink is changed
+    - update_uplink.current.0.vmmUplinkP.attributes.dn == "uni/vmmp-VMware/dom-ansible_domain/uplinkpcont/uplinkp-1"
+    - update_uplink.current.0.vmmUplinkP.attributes.uplinkId == "1"
+    - update_uplink.current.0.vmmUplinkP.attributes.uplinkName == "updated_uplink"
+
+# QUERY VMWARE UPLINK
+- name: Query VMWare Uplink
+  cisco.aci.aci_vmm_uplink:
+    <<: *aci_info
+    domain: ansible_domain
+    uplink_id: 1
+    state: query
+  register: query_uplink
+
+- name: Verify Uplink query
+  assert:
+    that:
+    - query_uplink is not changed
+    - query_uplink.current.0.vmmUplinkP.attributes.dn == "uni/vmmp-VMware/dom-ansible_domain/uplinkpcont/uplinkp-1"
+    - query_uplink.current.0.vmmUplinkP.attributes.uplinkId == "1"
+    - query_uplink.current.0.vmmUplinkP.attributes.uplinkName == "updated_uplink"
+
+# DELETE VMWARE UPLINK
+- name: Delete VMWare Uplink
+  cisco.aci.aci_vmm_uplink:
+    <<: *aci_info
+    domain: ansible_domain
+    uplink_id: 1
+    state: absent
+  register: remove_uplink
+
+- name: Verify VMWare Uplink deletion
+  assert:
+    that:
+    - remove_uplink is changed
+    - remove_uplink.current == []
+    - remove_uplink.previous.0.vmmUplinkP.attributes.dn == "uni/vmmp-VMware/dom-ansible_domain/uplinkpcont/uplinkp-1"
+    - remove_uplink.previous.0.vmmUplinkP.attributes.uplinkId == "1"
+    - remove_uplink.previous.0.vmmUplinkP.attributes.uplinkName == "updated_uplink"
+
+# DELETE VMWARE UPLINK AGAIN TO TEST IDEMPOTENCE
+- name: Delete VMWare Uplink again
+  cisco.aci.aci_vmm_uplink:
+    <<: *aci_info
+    domain: ansible_domain
+    uplink_id: 1
+    state: absent
+  register: remove_uplink_again
+
+- name: Verify VMWare Uplink deletion idempotence
+  assert:
+    that:
+    - remove_uplink_again is not changed
+
+# CLEAN UP
+- name: Remove ansible_domain
+  aci_domain:
+    <<: *aci_info
+    domain: ansible_domain
+    domain_type: vmm
+    vm_provider: vmware
+    state: absent
```

Comment:
This class is not available on cloud APICs and on versions prior to v4.2 so if you can add a block to limit it to non-cloud and version above 4.2.

Something like:
```
- name: Query system information
  aci_system:
    <<: *aci_info
    id: 1
    state: query
  register: version

- name: Verify Cloud and Non-Cloud Sites in use.
  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml

- name: Execute tasks only for non-cloud sites
  when: 
   - query_cloud.current == []  # This condition will skip execution for cloud sites
   - version.current.0.topSystem.attributes.version is version('4.2', '>=') # Make sure we execute it for APIC version above or equal to 4.2
  block:
```
Then move all of the next items inside the block statement (indentation)

Reply:
updated